### PR TITLE
Support to python3

### PR DESCRIPTION
**Pull request by alexandre**
All tests are passing, but I'm referencing some own repositories while I not receive a feedback about my others PR.

I've created a repository for the twill with support to python3 (2to3) and I would like to pass this repository to the quokkaproject profile, if you want...of course.

### CODE REVIEW

**Comment by ghost** [comment]
:+1:

**Comment by avelino** [comment]
+1

**Comment by rochacbruno** [comment]
:+1:  very nice! @alexandre I will take a closer look and merge soon!

**Comment by alexandre** [comment]
@rochacbruno ,

I'm updating the PR with the new feature (RSS) and keeping the python3 support. There are some PEP8 warnings, I'll fix it too, ok?

**Comment by ellisonleao** [comment]
nice one!

File: requirements.txt [hunk 1]
```diff
@@ -24,12 +24,12 @@ dealer
...
 flake8
 passlib
-twill
+https://github.com/alexandre/twill/tarball/master
 markupsafe
 Pillow
 pyyaml
 speaklater
-https://github.com/pythonhub/quokka-themes/tarball/master
+https://github.com/alexandre/quokka-themes/tarball/master
```
Comment (rochacbruno) [inline requirements.txt]: I merged your fork with quokkaproject/quokka-themes

File: requirements_test.txt [hunk 2]
```diff
@@ -32,8 +32,8 @@ markupsafe
 Pillow
 pyyaml
 speaklater
-https://github.com/pythonhub/quokka-themes/tarball/master
-https://github.com/pythonhub/flask-security/tarball/quokka
+https://github.com/alexandre/quokka-themes/tarball/master
+https://github.com/alexandre/flask-security/tarball/quokka
```
Comment (rochacbruno) [inline requirements_test.txt]: merged eith quokkaproject/flask-secutiry

File: requirements.txt [hunk 3]
```diff
@@ -24,12 +24,12 @@ dealer
 flask-collect
 flake8
 passlib
-twill
+https://github.com/alexandre/twill/tarball/master
```
Comment (rochacbruno) [inline requirements.txt:27]: are you going to submit it to original author, or create a pip module for your twill?

Comment (alexandre) [inline requirements.txt:27]: @rochacbruno ,

I'd like to pass this repo to the QuokkaProject  as a "fork" of the original. For now, I've sent an email to the author about to create a mirror on GitHub or something similar.

Comment (rochacbruno) [inline requirements.txt:27]: Nice! I will fork to QuokkaProject organization and add you as team member, so the best is to release it to PyPi with an alternative name "twill3"

Comment (rochacbruno) [inline requirements.txt:27]: https://github.com/quokkaproject/twill (also your are added to the organisation)

Comment (alexandre) [inline requirements.txt:27]: @rochacbruno ,

The author of the project twill (C. Titus Brown) sent me a answer: https://github.com/ctb/twill

I've sent another email asking him about plans to keep a python3 version on Pypi or a py2-py3 hybrid version.

File: quokka/core/fields.py [hunk 4]
```diff
@@ -40,7 +40,8 @@ def inner(*args, **kwargs):
 def _get(self):
     def inner(*args, **kwargs):
         values = only_matches(self, kwargs)
-        if len(values) > 1:
+        values = list(values)
+        if len(list(values)) > 1:
```
Comment (drgarcia1986) [inline quokka/core/fields.py]: We need to perform this cast twice?
Why not just?

``` python
values = list(values)
if len(values) > 1:
```

File: quokka/core/tests/test_models.py [hunk 5]
```diff
@@ -1,8 +1,13 @@
 # coding: utf-8
+import sys
+
 from . import BaseTestCase
 
 from ..models import Channel, Config, CustomValue
 
+if sys.version_info.major == 3:
+    unicode = lambda x: u'{}'.format(x)
```
Comment (drgarcia1986) [inline quokka/core/tests/test_models.py:9]: mapping to `str` is a good idea?:

``` python
if sys.version_info.major == 3:
    unicode = str
```

Comment (alexandre) [inline quokka/core/tests/test_models.py:9]: @drgarcia1986 ,

yeah, that's a better idea. =]

Comment (drgarcia1986) [inline quokka/core/tests/test_models.py:9]: Awesome :smile:

File: quokka/core/fields.py (hunk 4 above)
Comment (alexandre) [inline quokka/core/fields.py]: We should refactor this piece...I didn't see that because I did keep my focus on python3 support.
I'll fix this flags. =]

Comment (drgarcia1986) [inline quokka/core/fields.py]: Cool.
BTW is a great job, congratulations :clap: .

File: quokka/core/tests/test_models.py (hunk 5 above)
Comment (smurfix) [inline quokka/core/tests/test_models.py:9]: On 17.05.2015 17:22, Diego Garcia wrote:

> In quokka/core/tests/test_models.py
> https://github.com/quokkaproject/quokka/pull/190#discussion_r30471298:
> 
> >  from . import BaseTestCase
> > 
> >  from ..models import Channel, Config, CustomValue
> > 
> > +if sys.version_info.major == 3:
> > -    unicode = lambda x: u'{}'.format(x)
> 
> Awesome :smile:
> 
> Umm, why is that any different from "unicode = str" ?

## 

-- Matthias Urlichs

Comment (drgarcia1986) [inline quokka/core/tests/test_models.py:9]: I think is the better way (but the lambda approach is a good idea too)

http://lucumr.pocoo.org/2014/1/5/unicode-in-2-and-3/
_On the other hand on Python 2 we have two text types: str which for all intents and purposes is limited to ASCII + some undefined data above the 7 bit range, **unicode which is equivalent to the Python 3 str type** and one byte type bytearray which it inherited from Python 3._

File: requirements.txt (hunk 3 above)
Comment (rochacbruno) [inline requirements.txt:27]: NIce :) so you just have to send a PR direct to original repo... we can wait!

**Comment by rochacbruno** [comment]
@alexandre the twill author answered your email? 

we need to merge this PR soon, before code changes a lot.

I want to include your changes, only need to know what to do about twill and decide about the unicode issue above.

**Comment by alexandre** [comment]
@rochacbruno no email after that one. I think you can keep the repo on quokkaproject.

I stop [for a moment] my commits because personal problems (laptop broken). But I'll work on this today.

[ ]'s (hugs)

**Comment by alexandre** [comment]
updated

File: quokka/core/views.py [hunk 6]
```diff
@@ -60,11 +69,8 @@ def get(self, long_slug):
         user = get_current_user()
 
         if channel.roles:
-            forbidden = True
-            for role in user.roles:
-                if role in channel.roles:
-                    forbidden = False
-                    break  # break in first occurence
+            forbidden = next(
+                (True for role in user.roles if role in user.roles), False)
```
Comment (rochacbruno) [inline quokka/core/views.py:73]: I guess there is a problem here:

the loop should iterate `channel.roles`

Comment (rochacbruno) [inline quokka/core/views.py:73]: I will merge, and then try to think in a way to write a test for this feature.

**Pull request merged**